### ISSUE TITLE
Keep keypoints when moving to the next image

### ISSUE THREAD

**Issue by Mut1nyJD** [SkalskiP/make-sense#75]
For keypoint/landmark annotation I think it would be great if the editor area would keep the keypoints from the last image when moving to the next one. As most of the times you want the same number of keypoints but you have not only recreate all the points but also assign the labels this is very cumbersome when you have many points lets say more than 10. If you already have all the label assigned points you simply need then to move them to the right place this is a lot faster than clicking them again and assign labels especially if you have images that are very similar so the keypoint might only move by less than 20 pixels between images.

**Comment by SkalskiP** [comment]
Hi, @Mut1nyJD ! I am aware of the problem and it does not only concern landmarks but also, for example, the rects. However, my idea for a solution is quite different. What do you think about adding a mechanism for copying and pasting labels? I think it's a much more universal approach. Allows you to duplicate labels between slides but also within a single slide.

**Comment by Mut1nyJD** [comment]
@SkalskiP  I guess that would at least elevate the problem a bit yes but it still are at least two additional clicks or keyboard interactions you would have to do every time. When you annotate 100s of pictures each click you do notice.

**Comment by SkalskiP** [comment]
@Mut1nyJD, Of course you're right, every click when you have to label hundreds of photos is noticeable. But... Building such applications requires compromises. That's how I see it:
- At this point, you have to do all the work to label each photo. Create a landmark, set it, and select a label from the dropdown and so on..
- In the new solution, you could select multiple labels at once, simply with <kbd>Ctrl</kbd> + <kbd>A</kbd>, or with your mouse. Then <kbd>Ctrl</kbd> + <kbd>C</kbd>, you move on to the next image with <kbd>Ctrl</kbd> + <kbd>Right</kbd> and <kbd>Ctrl</kbd> + <kbd>V</kbd> to paste. In my opinion, it still simplifies the whole thing. 
- You should also think that the approach you have proposed is optimal for you, but maybe it is not necessarily optimal for all users.

**Comment by Mut1nyJD** [comment]
@SkalskiP 

Yes of course I have a biased view on the problem because I tried to use your tool for a current annotation task but abandoned it in favor of hacking together a solution of my own because the workflow is at least five times more efficient. The question you have to ask yourself with your tool what type of jobs are you trying to target. Are you expecting people to annotate a few tens, a few hundreds, a few thousands images or even tens of thousands of images. Like most tools I've tried in the past including makeSense fall apart if it is more than a few tens.

My suggestion was to make it optional via a tick box then a user can still decide to keep it or not. One click at the beginning of the session saving you thousands of interactions later on.

Sure your idea is fine but as I said think about annotating a thousand images and you have three thousands additional key combination presses

**Comment by Skiid** [comment]
> @Mut1nyJD, Of course you're right, every click when you have to label hundreds of photos is noticeable. But... Building such applications requires compromises. That's how I see it:
> 
> * At this point, you have to do all the work to label each photo. Create a landmark, set it, and select a label from the dropdown and so on..
> * In the new solution, you could select multiple labels at once, simply with Ctrl + A, or with your mouse. Then Ctrl + C, you move on to the next image with Ctrl + Right and Ctrl + V to paste. In my opinion, it still simplifies the whole thing.
> * You should also think that the approach you have proposed is optimal for you, but maybe it is not necessarily optimal for all users.

That function would come in very handy with any static camera setup where we only need to do smaller adjusmtents.

**Comment by SkalskiP** [comment]
Hi @Skiid 👋  ! If there is still demand for this feature, I will add it to the scpoe make-sense beta.

**Comment by Skiid** [comment]
@SkalskiP Thank you that feature is highly desired. At least for my workflow 😁.
